### DISCUSSION TITLE
Load assembly file on button event in dialog

### DISCUSSION THREAD

**Discussion by bhagatus** [oleg-shilo/wixsharp#1673] [Q&A] [answered]
Hello,

My requirement is to load assembly and call function when user clicks on Next button on dialog box. 

I have added _**_DefaultRefAssemblies_**_ in project declaration and trying to refer this assembly to create instance as **System.Reflection.Assembly.LoadFile("filename.dll")**

But it is giving error for access denied.

_"at System.Security.Util.StringExpressionSet.CreateListFromExpressions(String str, Boolean needFullPath)
   at System.Security.Permissions.FileIOPermission.AddPathList(FileIOPermissionAccess access, AccessControlActions control, String pathListOrig, Boolean checkForDuplicates, Boolean needFullPath, Boolean copyPathList)
   at System.Security.Permissions.FileIOPermission..ctor(FileIOPermissionAccess access, String path)
   at System.Reflection.Assembly.LoadFile(String path)"_

**Comment by oleg-shilo** [comment, marked as answer]
It means that the location of the assembly is wrong (and the message just incorrectly reports the error). Or indeed the permissions are not set as you expected. As you can see it's not WixSharp but CLR who throws the exception.

I suggest you do a simple experiment. Add the assembly as a `new File(...)` so your installer will deploy it to the target dir. Then in your click handler load the assembly from the known location and do your invoke.

I have also attached the sample project that shows you the technique:
![image](https://github.com/user-attachments/assets/793d4d05-8e8d-4ffc-bd6d-e38c1d100364)

[WixSharp Setup2.zip](https://github.com/user-attachments/files/17748302/WixSharp.Setup2.zip)

**Reply by bhagatus** [reply]
Hello Oleg,
Thank you for your inputs. After referring to sample project, I am able to solve the issue. Issue was with the method which we were using to get executing path of current assembly.  After using Appdomain path we are able to load the assembly.
